ssl/statem/statem_clnt.c

```diff
@@ -1105,7 +1105,9 @@ int tls_construct_client_hello(SSL *s, WPACKET *pkt)
         return 0;
     }
 #ifndef OPENSSL_NO_COMP
-    if (ssl_allow_compression(s) && s->ctx->comp_methods) {
+    if (ssl_allow_compression(s)
+            && s->ctx->comp_methods
+            && (SSL_IS_DTLS(s) || s->s3->tmp.max_ver < TLS1_3_VERSION)) {
```

Comment:
Well, I did think about that but ssl_allow_compression() is called by both the client and the server and the conditions are slightly different on each side. In the client case this is called before TLSv1.3 is actually selected (we are constructing the ClientHello so version negotiation hasn't happened yet). These conditions check whether the ClientHello is compatible with TLSv1.3. On the server side we check whether TLSv1.3 has actually been selected.